tests/test_json_model.py

```diff
@@ -290,6 +291,23 @@ def test_no_data_wrapper_deserialize(self):
         content = model.response(resp, content)
         self.assertEqual(content, {"data": "is good"})
 
+    def test_no_data_wrapper_deserialize_text_format(self):
+        model = JsonModel(data_wrapper=False)
+        resp = httplib2.Response({"status": "200"})
+        resp.reason = "OK"
+        content = CSV_TEXT_MOCK
+        content = model.response(resp, content)
+        self.assertEqual(content, CSV_TEXT_MOCK)
+
+    def test_no_data_wrapper_deserialize_raise_type_error(self):
+        buffer = io.StringIO()
+        buffer.write('String buffer')
+        model = JsonModel(data_wrapper=False)
+        resp = httplib2.Response({"status": "500"})
+        resp.reason = "The JSON object must be str, bytes or bytearray, not StringIO"
+        content = buffer
+        self.assertRaises(TypeError, model.response, (resp, content))
```

Comment:
We always prefer to use `assertRaises` as a context manager:
```suggestion
        with self.assertRaises(TypeError):
            model.response(resp, content)
```